src/psyclone/psyGen.py

```diff
@@ -4465,6 +4663,227 @@ def _number_handler(self, node, parent):
         return Literal(node.items[0], parent=parent)
 
 
+class Symbol(object):
+    '''
+    Symbol item for the Symbol Table. It contains information about: the name,
+    the datatype, the shape (in row-major order) and the symbol
+    access characterisctics. The symbol access can be:
+        - local: Variable that just exist in the kernel scope.
+        - external: Global variable.
+        - read_arg: Kernel argument which is only read.
+        - write_arg: Kernel argument which is only written.
+        - readwrite_arg: Kernel argument which is read and written.
+
+    :param str name: Name of the symbol.
+    :param str datatype: Data type of the symbol.
+    :param list shape: Shape of the symbol in row-major order (leftmost \
+                       index is contiguous in memory). Each entry represents \
+                       an array dimension. If not None then it holds the \
+                       extent of that dimension. If it is an empy list it \
+                       represents an scalar.
+    :param str access: String that specifies if the declaration \
+                       represents a 'local', 'external' or argument \
+                       variable, if it is an argument it also specifies \
+                       if it is 'read_arg', 'write_arg' or 'readwrite_arg'.
+    :raises NotImplementedError: Provided parameters are not supported yet.
+    :raises TypeError: Provided parameters have invalid error type.
+    :raises ValueError: Provided parameters contain invalid values.
+    '''
+
+    # Tuple with the valid values for the access attribute.
+    valid_access_types = ('local', 'external', 'read_arg', 'write_arg',
+                          'readwrite_arg')
+
+    def __init__(self, name, datatype=None, shape=[], access=None):
+
+        if datatype not in ('real', 'integer'):
+            raise NotImplementedError("Symbol can only be initialized with "
+                                      "'real' or 'integer' datatypes.")
+
+        if not isinstance(shape, list):
+            raise TypeError("Symbol shape attribute must be a list.")
+
+        if False in [isinstance(x, (type(None), int)) for x in shape]:
+            raise TypeError("Symbol shape list elements can only be "
+                            "'integer' or 'None'.")
+
+        if access not in Symbol.valid_access_types:
+            raise ValueError("Symbol access attribute can only be: "
```

Comment:
You can use `Symbol.valid_access_types` in this error message to future-proof it against any changes to the valid access types, e.g.:

    "...can only be one of '{0}' but got '{1}'".format(str(Symbol.valid_access_types), access)

Reply:
Neat